hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/endpoint/ObjectEndpoint.java

```diff
@@ -196,6 +199,15 @@ public Response put(
 
       // Normal put object
       OzoneBucket bucket = volume.getBucket(bucketName);
+      if (length == 0 &&
+          OZONE_S3G_FSO_DIRECTORY_CREATION_ENABLED &&
```

Comment:
Similar to above, this value needs to be checked for what the user set by getting the key from a `ConfigurationSource` or `OzoneConfiguration` object.